server/server.go

```diff
@@ -217,6 +218,9 @@ func (p *proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	case resp := <-pending.respChan:
 		defer resp.Body.Close()
+		for key, vals := range resp.Header {
+			w.Header()[key] = vals
```

Comment:
Do we want to forward all headers or just cookies?

Reply:
There's actually an RFC specifying which headers should be copied by proxies and which should be filtered (see https://www.w3.org/Protocols/rfc2616/rfc2616-sec13.html).

I was previously just being sloppy and copying everything, but I have now added the logic to distinguish between what should be copied and what should be filtered.